remove-api-key updated

jesusgonzalez12 opened the pull request:
This version introduces improvements on how a user can remove an API key using the pyc8 SDK.
Only changes to the pyc8 code have been introduced.
For more details, you can check the Kanban board.

jesusgonzalez12 committed:
remove-api-key updated

This version introduces improvements on how a user can remove an API key using the pyc8 SDK.
Only changes to the pyc8 code have been introduced.
For more details, you can check the Kanban board.

jamesm-macrometa commented on docs/sdks/remove-api-key.md:
```diff
@@ -32,10 +32,23 @@ client = C8Client(protocol='https', host='gdn.paas.macrometa.io', port=443,
+        break
 
-print(remove)
+# Checks if the API already exists and acts accordingly
```
```suggestion
# Checks the specified API key
```

If I understand correctly, this step merely checks the API key, and the next one removes it. Correct?

jamesm-macrometa commented on docs/sdks/remove-api-key.md:
```diff
@@ -32,10 +32,23 @@ client = C8Client(protocol='https', host='gdn.paas.macrometa.io', port=443,
+        print("Error removing the API key")
+        print(err)
+else:
+    # Display message that the eky does not exists
```
```suggestion
    # Displays a message that the key does not exist
```

jamesm-macrometa commented on docs/sdks/remove-api-key.md:
```diff
@@ -32,10 +32,23 @@ client = C8Client(protocol='https', host='gdn.paas.macrometa.io', port=443,
-print(remove)
+# Checks if the API already exists and acts accordingly
+if API_ACTIVE:
+    # Remove the chosen API key
```
```suggestion
    # Removes the chosen API key
```

jamesm-macrometa commented on docs/sdks/remove-api-key.md:
```diff
@@ -32,10 +32,23 @@ client = C8Client(protocol='https', host='gdn.paas.macrometa.io', port=443,
 
-# Remove api key
-remove = client.remove_api_key('id1')
+# Check wether the chosen_api_key exists or not, comparing it to the list elements
```
```suggestion
# Checks if `chosen_api_key` exists
```

What do we mean by "comparing it to the list elements"?

jesusgonzalez12 replied:
Sorry, I missed adding a few lines of code. Fixed.

A list of all API key IDs a user has is saved in a variable which is used to make a comparison check later.

jesusgonzalez12 replied:
I have updated the comment to improve semantics.

jamesm-macrometa reviewed (changes requested):
A few minor changes, but looks good!

jesusgonzalez12 committed:
Apply suggestions from code review

Co-authored-by: James <86624594+jamesm-macrometa@users.noreply.github.com>

jesusgonzalez12 committed:
Added missing code about element list

Added missing code where a list of API key IDs is saved in a variable for later comparison.

jesusgonzalez12 commented:
Thanks for the review.
Please, check again the element list topic.

jesusgonzalez12 committed:
Updated semmantics on a comment

Updated a few words in a commented line to improve semmantics.

diana-macrometa commented on docs/sdks/remove-api-key.md:
```diff
@@ -31,11 +31,29 @@ from c8 import C8Client
 
-# Remove api key
-remove = client.remove_api_key('id1')
+# Check wether the chosen_api_key exists or not, comparing it to the API key list
```
```suggestion
# Check whether the chosen_api_key exists or not, comparing it to the API key list
```

diana-macrometa commented on docs/sdks/remove-api-key.md:
```diff
@@ -31,11 +31,29 @@ from c8 import C8Client
+if API_ACTIVE:
+    # Removes the chosen API key
+    try:
+        print("Remove API Key: ", client.remove_api_key(CHOSEN_API_KEY_ID))
```
```suggestion
        print("Remove API key: ", client.remove_api_key(CHOSEN_API_KEY_ID))
```

jesusgonzalez12 replied:
Thanks for the suggestions @diana-macrometa. Updated with suggestions.

jesusgonzalez12 committed:
Apply suggestions from code review

Co-authored-by: Diana Payton <105658419+diana-macrometa@users.noreply.github.com>

diana-macrometa commented on docs/sdks/remove-api-key.md:
```diff
@@ -31,11 +31,29 @@ from c8 import C8Client
 
-# Remove api key
-remove = client.remove_api_key('id1')
+# Check whether the chosen_api_key exists or not, comparing it to the API key list
```
Why is chosen_api_key lowercase here and uppercase in line 42?

jesusgonzalez12 replied:
Hi Diana. PEP8 standard (and JS standard is similar) states that constants should be in upper-case.

We recently changed our code to be as PEP8 compliant as possible.
And certainly, that comment line was bypassed while making the conversion.
Will fix it shortly.

diana-macrometa commented on docs/sdks/remove-api-key.md:
```diff
@@ -31,11 +31,29 @@ from c8 import C8Client
+        print(err)
+else:
+    # Displays a message that the key does not exist
+    print("Can't remove a non-existent key")
```
```suggestion
    print("Can't remove a non-existent key.")
```

diana-macrometa replied:
Be sure to finish sentences with periods. Apply throughout.

jesusgonzalez12 replied:
Thanks Diana.

This writing style is not in the original files.
Good to have it as a way to increase consistency on the docs.
Fixed.

diana-macrometa committed:
Merge branch 'master' into patch-4

diana-macrometa commented:
@jesusgonzalez12, click the circling arrows next to a reviewer to request another review.

jesusgonzalez12 committed:
Suggestions applied

jamesm-macrometa commented on docs/sdks/remove-api-key.md:
```diff
@@ -31,11 +31,29 @@ from c8 import C8Client
 
-# Remove api key
-remove = client.remove_api_key('id1')
+# Check whether the CHOSEN_API_KEY_ID exists or not, comparing it to the API key list
```
```suggestion
# Checks if the API key exists
```

jesusgonzalez12 committed:
Update docs/sdks/remove-api-key.md

Co-authored-by: James <86624594+jamesm-macrometa@users.noreply.github.com>

koshy-thomas commented on docs/sdks/remove-api-key.md:
```diff
@@ -31,11 +31,29 @@ from c8 import C8Client
+# Checks if the API key exists
+for apiElement in apiKeyIds:
+    if apiElement == CHOSEN_API_KEY_ID:
+        API_ACTIVE = True
```
@jesusgonzalez12 I guess you forgot to declare CHOSEN_API_KEY_ID and API_ACTIVE on top after import C8Client.
Example:
API_ACTIVE = False
CHOSEN_API_KEY_ID = "YourApiKey"

koshy-thomas replied:
@jesusgonzalez12 The same is missing I guess in "docs/sdks/create-api-key.md" as well you can add the declaration of these two variables there too.

jesusgonzalez12 replied:
Thanks @koshy-thomas. You're right, thanks for checking. Making changes.

PS: for this I have a single file that adds the key if the ID does not exists while removing it If the key already exists. So translating that to two separate files I had to change a few prints and picky-choose what I needed for each example. As I can see, I have room for improvement on that department.

Changes here are complete. Will now check PR 205.

jesusgonzalez12 committed:
Applying suggestions

Applying latest suggestions

koshy-thomas committed:
Merge branch 'master' into patch-4
